### DISCUSSION TITLE
About Login, How can I make a guest login or anonymous login (use first then login), please ? 请问如何进行访客登录或匿名登录（先使用后登录）?

### DISCUSSION THREAD

**Discussion by KeeeevinK** [lobehub/lobehub#4844] [Self Hosting]
请问使用哪一个功能可以让朋友简化登录，最好是可以先直接使用后续需要的时候再登录，如果不行那么使用clerk还是nextauth中的哪一个验证方式最简便呢？（第一次配置登录功能，阅读了开发文档依旧不是很清楚🙏🥲）Docker
Which feature can allow friends to simplify the login process, preferably to use directly first and then log in later when needed? If not possible, which authentication method between Clerk and NextAuth is the most convenient? (I'm configuring the login feature for the first time and still not quite clear even after reading the development documentation. 🙏🥲 thanks)  Docker

Ready to use Docker or Compose deploy

**Comment by SpeedupMaster** [comment]
vercel部署用clerk比较简单，docker部署用nextauth，docker部署如果想用clerk的话，需要自己构建镜像，麻烦点。

**Reply by KeeeevinK** [reply]
请问 nextauth 下的验证方式没有支持访客登录的吗？或者是不登录

**Reply by SpeedupMaster** [reply]
> 请问 nextauth 下的验证方式没有支持访客登录的吗？或者是不登录

好像都没有访客登录吧

**Reply by KeeeevinK** [reply]
clerk 是可以访客登录吗？我看了下官方的web 好像也没有这个选项诶

**Reply by SpeedupMaster** [reply]
> clerk 是可以访客登录吗？我看了下官方的web 好像也没有这个选项诶

clerk好像也没有访客登录。这些认证好像都没有这个功能

**Reply by KeeeevinK** [reply]
好吧🥲，那我应该用那个身份验证简单一些？国内朋友用最好不用接验证码，看到有 logto 做的填账号和密码就行不过不能谷歌账号验证

**Reply by SpeedupMaster** [reply]
> 好吧🥲，那我应该用那个身份验证简单一些？国内朋友用最好不用接验证码，看到有 logto 做的填账号和密码就行不过不能谷歌账号验证

基本上都可以直接用账户密码登录，验证码不设置就行

**Reply by KeeeevinK** [reply]
还想请教一下，logto 的本地部署了好像还是要走外网吗？ 佬你是配的 zitadel 吗，有什么好吗它的 demo 我都没找到

**Comment by cy948** [comment]
docker compose 部署的一键部署脚本默认使用 casdoor 登录，该登录只需要简单填写账户密码进行注册，不需要邮箱地址，接近于匿名登录。

**Reply by KeeeevinK** [reply]
目前是考虑用这个了，但是跳转的时候会失败，可能是我是 cftunnel 的问题吗？
还有个是 UI 没有 clerk 和 logto 好看，是不是只有 clerk 可以做小窗登录（不跳转的登录）

**Reply by SpeedupMaster** [reply]
> 目前是考虑用这个了，但是跳转的时候会失败，可能是我是 cftunnel 的问题吗？ 还有个是 UI 没有 clerk 和 logto 好看，是不是只有 clerk 可以做小窗登录（不跳转的登录）

casdoor的UI确实有点丑，还可以试试zitadel，UI也还行。除了clerk外其他的好像都要跳转，用了一圈感觉还是clerk好用，现在自己构建镜像用clerk了

**Reply by Richardallis411** [reply]
Release

**Reply by KeeeevinK** [reply]
> > 目前是考虑用这个了，但是跳转的时候会失败，可能是我是 cftunnel 的问题吗？ 还有个是 UI 没有 clerk 和 logto 好看，是不是只有 clerk 可以做小窗登录（不跳转的登录）
> 
> casdoor的UI确实有点丑，还可以试试zitadel，UI也还行。除了clerk外其他的好像都要跳转，用了一圈感觉还是clerk好用，现在自己构建镜像用clerk了
> 
> 

牛哇， clerk 国内环境可以用吗？还是要在服务器配代理，客人就能用

**Reply by SpeedupMaster** [reply]
> 牛哇， clerk 国内环境可以用吗？还是要在服务器配代理，客人就能用

我没配置代理，可以直接用

**Reply by KeeeevinK** [reply]
> > 目前是考虑用这个了，但是跳转的时候会失败，可能是我是 cftunnel 的问题吗？ 还有个是 UI 没有 clerk 和 logto 好看，是不是只有 clerk 可以做小窗登录（不跳转的登录）
> 
> casdoor的UI确实有点丑，还可以试试zitadel，UI也还行。除了clerk外其他的好像都要跳转，用了一圈感觉还是clerk好用，现在自己构建镜像用clerk了
> 
> 

佬有空可以出个评测贴吗，登录这方面文档的介绍真的很少，没配过的一开始就不懂

**Reply by SpeedupMaster** [reply]
> > > 目前是考虑用这个了，但是跳转的时候会失败，可能是我是 cftunnel 的问题吗？ 还有个是 UI 没有 clerk 和 logto 好看，是不是只有 clerk 可以做小窗登录（不跳转的登录）
> > 
> > 
> > casdoor的UI确实有点丑，还可以试试zitadel，UI也还行。除了clerk外其他的好像都要跳转，用了一圈感觉还是clerk好用，现在自己构建镜像用clerk了
> 
> 佬有空可以出个评测贴吗，登录这方面文档的介绍真的很少，没配过的一开始就不懂

vercel部署那部分有clerk的配置教程，按照来就行了

**Reply by KeeeevinK** [reply]
> > > > 目前是考虑用这个了，但是跳转的时候会失败，可能是我是 cftunnel 的问题吗？ 还有个是 UI 没有 clerk 和 logto 好看，是不是只有 clerk 可以做小窗登录（不跳转的登录）
> > > 
> > > 
> > > casdoor的UI确实有点丑，还可以试试zitadel，UI也还行。除了clerk外其他的好像都要跳转，用了一圈感觉还是clerk好用，现在自己构建镜像用clerk了
> > 
> > 佬有空可以出个评测贴吗，登录这方面文档的介绍真的很少，没配过的一开始就不懂
> 
> vercel部署那部分有clerk的配置教程，按照来就行了

docker 有办法吗？vercel 不会用..

**Reply by SpeedupMaster** [reply]
> > > > > 目前是考虑用这个了，但是跳转的时候会失败，可能是我是 cftunnel 的问题吗？ 还有个是 UI 没有 clerk 和 logto 好看，是不是只有 clerk 可以做小窗登录（不跳转的登录）
> > > > 
> > > > 
> > > > casdoor的UI确实有点丑，还可以试试zitadel，UI也还行。除了clerk外其他的好像都要跳转，用了一圈感觉还是clerk好用，现在自己构建镜像用clerk了
> > > 
> > > 
> > > 佬有空可以出个评测贴吗，登录这方面文档的介绍真的很少，没配过的一开始就不懂
> > 
> > 
> > vercel部署那部分有clerk的配置教程，按照来就行了
> 
> docker 有办法吗？vercel 不会用..

docker除了要自己构建镜像，clerk配置和vercel基本上一样

**Reply by KeeeevinK** [reply]
等你们的镜像了，logto 好多 bug，
还发现一个叫 authing， 是国产 auth0 的，免费版有 8000 人额度

**Reply by KeeeevinK** [reply]
> > > > > > 目前是考虑用这个了，但是跳转的时候会失败，可能是我是 cftunnel 的问题吗？ 还有个是 UI 没有 clerk 和 logto 好看，是不是只有 clerk 可以做小窗登录（不跳转的登录）
> > > > > 
> > > > > 
> > > > > casdoor的UI确实有点丑，还可以试试zitadel，UI也还行。除了clerk外其他的好像都要跳转，用了一圈感觉还是clerk好用，现在自己构建镜像用clerk了
> > > > 
> > > > 
> > > > 佬有空可以出个评测贴吗，登录这方面文档的介绍真的很少，没配过的一开始就不懂
> > > 
> > > 
> > > vercel部署那部分有clerk的配置教程，按照来就行了
> > 
> > docker 有办法吗？vercel 不会用..
> 
> docker除了要自己构建镜像，clerk配置和vercel基本上一样

https://www.authing.co/solutions/auth0

**Reply by SpeedupMaster** [reply]
> > > > > > 目前是考虑用这个了，但是跳转的时候会失败，可能是我是 cftunnel 的问题吗？ 还有个是 UI 没有 clerk 和 logto 好看，是不是只有 clerk 可以做小窗登录（不跳转的登录）
> > > > > 
> > > > > 
> > > > > casdoor的UI确实有点丑，还可以试试zitadel，UI也还行。除了clerk外其他的好像都要跳转，用了一圈感觉还是clerk好用，现在自己构建镜像用clerk了
> > > > 
> > > > 
> > > > 佬有空可以出个评测贴吗，登录这方面文档的介绍真的很少，没配过的一开始就不懂
> > > 
> > > 
> > > vercel部署那部分有clerk的配置教程，按照来就行了
> > 
> > 
> > docker 有办法吗？vercel 不会用..
> 
> docker除了要自己构建镜像，clerk配置和vercel基本上一样

https://github.com/lobehub/lobe-chat/blob/main/Dockerfile.database
```
ARG NEXT_PUBLIC_CLERK_PUBLISHABLE_KEY
ARG CLERK_SECRET_KEY
ARG CLERK_WEBHOOK_SECRET
ENV NEXT_PUBLIC_CLERK_PUBLISHABLE_KEY="pk_live_Y2xlcmxxxxxxxx" \
    CLERK_SECRET_KEY="" \
    CLERK_WEBHOOK_SECRET=""
```
要使用你自己的 `NEXT_PUBLIC_CLERK_PUBLISHABLE_KEY` 只能自己构建镜像，在Dockerfile文件中添加上面的几行，然后自己构建docker镜像
